torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -37,37 +37,41 @@
 
 
 @no_type_check
-def _validate_hybrid_shard_setup(fsdp_root: _FSDPState, fsdp_module: nn.Module):
+def _validate_and_get_hybrid_shard_state(root_module: nn.Module):
     """
-    Performs validation that hybrid sharding strategy is setup. In particular, we:
-    1) Ensure root and passed in FSDP module have the same hybrid sharding strategy,
-    i.e. both should be using the same hybrid shard strategy or no hybrid shard at all.
-    2) Ensure that inter and intra-node process groups are the same across root and
-    this FSDP module.
-    """
-    if (
-        fsdp_root.sharding_strategy in HYBRID_SHARDING_STRATEGIES
-        or fsdp_module.sharding_strategy in HYBRID_SHARDING_STRATEGIES
-    ):
-        if fsdp_root.sharding_strategy != fsdp_module.sharding_strategy:
-            raise ValueError(
-                "When using hybrid sharding strategy, expect sharding strategies"
-                f" to be the same, but got {fsdp_root.sharding_strategy} vs {fsdp_module.sharding_strategy}"
-            )
+    Precondition: ``root_module`` is a ``FullyShardedDataParallel`` instance.
 
-        # Ensure inter and intra-node process groups are the same
-        # TODO (rohan-varma) unclear whether these should be asserts or Exceptions
-        # as they can happen due to bug in FSDP process group setup or user passing in
-        # incorrect configuration.
-        if fsdp_root.process_group != fsdp_module.process_group:
-            raise ValueError(
-                f"For {fsdp_root.sharding_strategy} intra-node process groups do not match"
-            )
+    This checks that all instances using a hybrid sharding strategy have the
+    same intra- and inter-node process groups.
 
-        if fsdp_root._inter_node_pg != fsdp_module._inter_node_pg:
-            raise ValueError(
-                f"For {fsdp_root.sharding_strategy}, inter-node process groups do not match"
-            )
+    Returns:
+        Tuple[HookState]: One of the instances' inter-node state (does not
```

Comment:
curious, I don't see `HookState` anywhere?

Reply:
Good catch. I recalled it being called that, but it is actually called `DefaultState`. We may want to revisit that name in the future.